Code/Tools/Libs/GuiFoundation/NodeEditor/Implementation/Connection.cpp

```diff
@@ -168,4 +186,102 @@ void ezQtConnection::paint(QPainter* painter, const QStyleOptionGraphicsItem* op
   }
 
   painter->drawPath(path());
+
+  if (decorationFlags.IsSet(ezQtNodeScene::ConnectionDecorationFlags::DrawDebugging))
+  {
+    const float offset = fmod(ezTime::Now().GetSeconds(), 1.0f);
+    const qreal segments = path().length() / 16;
+
+    for (qreal length = 0; length < segments + 0.0005f; ++length)
+    {
+      painter->drawEllipse(path().pointAtPercent(path().percentAtLength((length + offset) * 16)), 2, 2);
+    }
+  }
+}
+
+void ezQtConnection::DrawSubwayPath(QPainterPath& path, const QPointF& startPoint, const QPointF& endPoint)
+{
+  const bool isStartAboveTarget = startPoint.y() <= endPoint.y();
```

Comment:
This looks quite involved. Can you explain what a "subway path" does?

Reply:
Ill add a comment for it.
It draws the same way you would see on many subway maps. this means only straight and diagonal lines to reach the end goal. It also handles connections to nodes to the left of the output 

![Subway](https://github.com/ezEngine/ezEngine/assets/3044415/fbd21b2c-e413-4a19-ad06-6ff93cd0d881)